src/compiler/checker.ts

```diff
@@ -36923,10 +36923,14 @@ namespace ts {
             const tracingData: tracing.EventData = [tracing.Phase.Check, "checkSourceFile", { path: node.path }];
             tracing.begin(...tracingData);
             performance.mark("beforeCheck");
-            checkSourceFileWorker(node);
-            performance.mark("afterCheck");
-            performance.measure("Check", "beforeCheck", "afterCheck");
-            tracing.end(...tracingData);
+            try {
+                checkSourceFileWorker(node);

```

Comment:
I still dont like the fact that you have to add try finally here. why cant existing try catch finally block underwhich this dunction always runs handle this. We should be able to rever the stack to that place so that we dont have to worry about try catch blocks in future either

Reply:
I have to admit that the universal backlash against try blocks caught me off-guard. 😄 

https://github.com/microsoft/TypeScript/pull/41498 should make it possible to eliminate them.  Once that PR is merged, I'll rebase this one.

Reply:
Just to repeat it here, I have withdrawn my objection to it, on the assumption that `try-finally` has negligible cost (unlike `-catch`).